haystack/utils/openai_utils.py

```diff
@@ -128,7 +128,8 @@ def _openai_text_completion_tokenization_details(model_name: str):
 
 
 @tenacity.retry(
-    retry=tenacity.retry_if_exception_type(OpenAIRateLimitError),
+    reraise=True,
+    retry=tenacity.retry_if_exception_type((OpenAIRateLimitError, OpenAIError)),
```

Comment:
I think specifying here only `OpenAIError` should be enough, as `OpenAIRateLimitError` inherits from `OpenAIError`.